lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,993 @@
+    - "boto3"
+version_added: "2.4"
+author:
+    - Bruce Pennypacker (@bpennypacker)
+    - Will Thames (@willthames)
+    - Michael De La Rue (@mikedlr)
+options:
+  state:
+    description:
+      - Describes the desired state of the database instance. N.B. restarted is allowed as an alias for rebooted.
+    required: false
+    default: present
+    choices: [ 'present', 'absent', 'rebooted' ]
+  db_instance_identifier:
+    description:
+      - Database instance identifier.
+    required: true
+  source_instance:
+    description:
+      - Name of the database when sourcing from a replica
+    required: false
+  replica:
+    description:
+    - whether or not a database is a read replica
+    default: False
+  engine:
+    description:
+      - The type of database. Used only when state=present.
+    required: false
+    choices: [ 'mariadb', 'MySQL', 'oracle-se1', 'oracle-se2', 'oracle-se', 'oracle-ee', 'sqlserver-ee',
+                sqlserver-se', 'sqlserver-ex', 'sqlserver-web', 'postgres', 'aurora']
+  allocated_storage:
+    description:
+      - Size in gigabytes of the initial storage for the DB instance.  See [API documentation](https://botocore.readthedocs.io/en/latest/reference/services/rds.html#RDS.Client.create_db_instance) for details of limits
+      - Required unless the database type is aurora,
+  storage_type:
+    description:
+      - Specifies the storage type to be associated with the DB instance. C(iops) must
+        be specified if C(io1) is chosen.
+    choices: ['standard', 'gp2', 'io1' ]
+    default: standard unless iops is set
+  db_instance_class:
+    description:
+      - The instance type of the database. If source_instance is specified then the replica inherits
+        the same instance type as the source instance.
+  username:
+    description:
+      - Master database username.
+  password:
+    description:
+      - Password for the master database username.
+  db_name:
+    description:
+      - Name of a database to create within the instance. If not specified then no database is created.
+  engine_version:
+    description:
+      - Version number of the database engine to use. If not specified then
+      - the current Amazon RDS default engine version is used.
+  parameter_group:
+    description:
+      - Name of the DB parameter group to associate with this instance. If omitted
+      - then the RDS default DBParameterGroup will be used.
+    required: false
+  license_model:
+    description:
+      - The license model for this DB instance.
+    required: false
+    choices:  [ 'license-included', 'bring-your-own-license', 'general-public-license', 'postgresql-license' ]
+  multi_zone:
+    description:
+      - Specifies if this is a Multi-availability-zone deployment. Cazn not be used in conjunction with zone parameter.
+    choices: [ "yes", "no" ]
+    required: false
+  iops:
+    description:
+      - Specifies the number of IOPS for the instance. Must be an integer greater than 1000.
```

Comment:
That's what the AWS docs say.  This does come under our discussion during the working group about where we should use reflection / catching exceptions rather than hard wiring constants into modules.